src/compilation.ts

```diff
@@ -0,0 +1,655 @@
+import {
+  print,
+  visit,
+  visitWithTypeInfo,
+  typeFromAST,
+  getNamedType,
+  isAbstractType,
+  isEqualType,
+  isTypeSubTypeOf,
+  doTypesOverlap,
+  Kind,
+  TypeInfo,
+  isType,
+  isCompositeType,
+  GraphQLScalarType,
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+  GraphQLObjectType,
+  GraphQLInterfaceType,
+  GraphQLUnionType,
+  GraphQLString,
+  GraphQLInt,
+  GraphQLFloat,
+  GraphQLBoolean,
+  GraphQLID,
+  GraphQLError,
+  GraphQLSchema,
+  GraphQLType,
+  GraphQLCompositeType,
+  DocumentNode,
+  OperationDefinitionNode,
+  FragmentDefinitionNode,
+  VariableNode,
+  SelectionSetNode,
+  FieldNode,
+  ArgumentNode,
+  DirectiveNode
+} from 'graphql';
+
+import {
+  isTypeProperSuperTypeOf,
+  getOperationRootType,
+  getFieldDef,
+  valueFromValueNode,
+  filePathForNode,
+  sourceAt,
+  withTypenameFieldAddedWhereNeeded,
+  isBuiltInScalarType
+} from './utilities/graphql';
+
+import { join, block, wrap, indent } from './utilities/printing';
+
+import {
+  createHash,
+} from 'crypto';
+
+export interface CompilerOptions {
+  addTypename?: boolean;
+  mergeInFieldsFromFragmentSpreads?: boolean;
+  passthroughCustomScalars?: boolean;
+  customScalarsPrefix?: string;
+  namespace?: string;
+  generateOperationIds?: boolean;
+}
+
+export interface CompilationContext {
+  schema: GraphQLSchema;
+  operations: { [operationName: string]: CompiledOperation };
+  fragments: { [fragmentName: string]: CompiledFragment };
+  typesUsed: GraphQLType[];
+  options: CompilerOptions;
+}
+
+export interface CompiledOperation {
+  filePath?: string;
+  operationName: string;
+  operationId?: string;
+  operationType: string;
+  rootType: GraphQLObjectType;
+  variables: {
+    name: string;
+    type: GraphQLType;
+  }[];
+  source: string;
+  sourceWithFragments?: string;
+  fields: Field[];
+  fragmentSpreads?: string[];
+  inlineFragments?: CompiledInlineFragment[];
+  fragmentsReferenced: string[];
+}
+
+export interface CompiledFragment {
+  filePath?: string;
+  fragmentName: string;
+  source: string;
+  typeCondition: GraphQLCompositeType;
+  possibleTypes: GraphQLObjectType[];
+  fields: Field[];
+  fragmentSpreads: string[];
+  inlineFragments: any[];
```

Comment:
Ah, it turned out I had only changed it in one location but forgot the other one. Updated.